### ISSUE TITLE
[Feature]-[658]:Add Alt-Text property for Image widget

### ISSUE THREAD

**Issue by GreenFlux** [appsmithorg/appsmith#13383]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

This feature request is for an `Alt-Text` property for all widgets, which will display when the widget's visibility is set to hidden. The value could also be used to provide Alt-Text to the browser for Image widgets, and improve Appsmith's accessibility. 

### Why should this be worked on?

When a widget is set to hidden, the canvas is empty in that area and nothing else can be displayed in its place to communicate to the user why a certain feature/function is missing in their app. Another widget can be used to display a message to the user, but since widgets can't be stacked it just takes up more 'real estate' in the page.  The empty space from the hidden widget would be the ideal place for this type of message to the user.

**Comment by Nikhil-Nandagopal** [comment]
+1 it should only be for image widgets

**Comment by GreenFlux** [comment]
> @GreenFlux how is Alt-text relevant to all the widgets, I see the use case for image widget only right now

Actually, I'm requesting an alternative text to be displayed for ANY widget, when that widget is set to hidden. Since that leaves an empty place on the canvas, this seems like the ideal place to display a message about why any type of widget has been hidden. 

I'm suggesting that this feature could also be used for images- in the more traditional alt-text way, but really it's for all widgets when they have been hidden by a JS condition and we want to provide a message to the user it its place. 

Perhaps calling it alt-text is confusing. Maybe this should be called 'Hidden Message' instead. But I think this feature would be useful for all widgets, which is my primary request. The use of this text as an image's alt-text is more of an added bonus for this feature. 

**Use Cases:** 
- Button has been hidden by JS because conditions are not met. Alt-Text displays explanation to user, prompting to fill out required form. 
- Child table is hidden because a row has not been selected in the parent table. Message can be displayed in place of child table while it is hidden, prompting user to select row on parent table. 

The empty space on the canvas is the most logical place to display a message about why a widget has been hidden, but we can't stack widgets. If we had a property to provide an alternative text it its place then any hidden widget could inform the user of what conditions are causing the feature to be disabled.

**Comment by Nikhil-Nandagopal** [comment]
@GreenFlux web apps don't usually have hidden messages right? The request seems more like a patch because of the whitespace that is occupied when actually we should simply solve the problem that if a widget is not visible, it should not take up space. What do you think?

**Comment by Nikhil-Nandagopal** [comment]
@GreenFlux yes we're working on enabling that as part of the dynamic height project

**Comment by vhemery** [comment]
I would like to contribute a PR which adds an "alt" text property to the image widget (with tests).
This PR addresses only the "alt" property for accessibility, to display in the alt HTML attribute. It does nothing when widget is hidden and has nothing to do with the dynamic height project.

@Nikhil-Nandagopal 
Should I link this PR to this feature issue or assign it to another new feature issue?
